usage/app/lib/ContentPollStream.scala

```diff
@@ -107,8 +115,28 @@ trait ContentPollStream {
   }
 
   val pollInterval  = Duration(pollIntervalInSeconds, SECONDS)
-  val rawObservable =  Observable.interval(pollInterval)
-    .flatMap(_ => getContent)
+  case class PollAttempt(lastFail: Throwable, i: Int)
+
+  lazy val retryingPollObservable = Observable.interval(pollInterval)
+    .flatMap(_ => getContent).retryWhen(
+      _.zipWith(Observable.from(1 to maxRetries))((e: Throwable, i: Int) => PollAttempt(e,i))
+      .flatMap(attempt => {
```

Comment:
Note: you can destructure in the flatMap to avoid having to dereference properties off the object every time:

``` scala
  flatMap { case PollAttempt(lastFail, i) =>
    // can use i and lastFail
  }
```